server/odc-service/src/main/java/com/oceanbase/odc/service/connection/logicaldatabase/LogicalDatabaseService.java

```diff
@@ -0,0 +1,228 @@
+import com.oceanbase.odc.metadb.connection.ConnectionEntity;
+import com.oceanbase.odc.metadb.connection.DatabaseEntity;
+import com.oceanbase.odc.metadb.connection.DatabaseRepository;
+import com.oceanbase.odc.metadb.connection.logicaldatabase.DatabaseMappingEntity;
+import com.oceanbase.odc.metadb.connection.logicaldatabase.DatabaseMappingRepository;
+import com.oceanbase.odc.metadb.connection.logicaldatabase.TableMappingRepository;
+import com.oceanbase.odc.metadb.dbobject.DBObjectRepository;
+import com.oceanbase.odc.service.collaboration.environment.EnvironmentService;
+import com.oceanbase.odc.service.collaboration.environment.model.Environment;
+import com.oceanbase.odc.service.connection.database.DatabaseService;
+import com.oceanbase.odc.service.connection.database.model.Database;
+import com.oceanbase.odc.service.connection.database.model.DatabaseSyncStatus;
+import com.oceanbase.odc.service.connection.database.model.DatabaseType;
+import com.oceanbase.odc.service.connection.logicaldatabase.model.CreateLogicalDatabaseReq;
+import com.oceanbase.odc.service.connection.logicaldatabase.model.DetailLogicalDatabaseResp;
+import com.oceanbase.odc.service.db.schema.model.DBObjectSyncStatus;
+import com.oceanbase.odc.service.iam.ProjectPermissionValidator;
+import com.oceanbase.odc.service.iam.auth.AuthenticationFacade;
+import com.oceanbase.odc.service.permission.DBResourcePermissionHelper;
+import com.oceanbase.odc.service.permission.database.model.DatabasePermissionType;
+
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * @Author: Lebie
+ * @Date: 2024/5/8 17:33
+ * @Description: []
+ */
+@Service
+@SkipAuthorize
+@Slf4j
+public class LogicalDatabaseService {
+
+    @Autowired
+    private ProjectPermissionValidator projectPermissionValidator;
+
+    @Autowired
+    private DatabaseRepository databaseRepository;
+
+    @Autowired
+    private DatabaseMappingRepository databaseMappingRepository;
+
+    @Autowired
+    private DBObjectRepository dbObjectRepository;
+
+    @Autowired
+    private TableMappingRepository tableMappingRepository;
+
+    @Autowired
+    private ConnectionConfigRepository connectionRepository;
+
+    @Autowired
+    private AuthenticationFacade authenticationFacade;
+
+    @Autowired
+    private EnvironmentService environmentService;
+
+    @Autowired
+    private DatabaseService databaseService;
+
+    @Autowired
+    private LogicalTableService tableService;
+
+    @Autowired
+    private LogicalDatabaseSyncManager syncManager;
+
+    @Autowired
+    private DBResourcePermissionHelper permissionHelper;
+
+
+    @Transactional(rollbackFor = Exception.class)
+    public Boolean create(CreateLogicalDatabaseReq req) {
+        preCheck(req);
+
+        Long organizationId = authenticationFacade.currentOrganizationId();
+        DatabaseEntity basePhysicalDatabase = databaseRepository
+                .findById(req.getPhysicalDatabaseIds().iterator().next()).orElseThrow(() -> new NotFoundException(
```

Comment:
Only care about the first element?

Reply:
Yes, we expect that all physical databases have the same environment, dialect type and we check it in `preCheck`. So we just need use the first database as the base database.